warden/app/oracle.go

```diff
@@ -0,0 +1,242 @@
+				// and extend our vote based on these prices
+				currencypair.NewDeltaCurrencyPairStrategy(app.OracleKeeper),
+			),
+			app.OracleKeeper,
+			veCodec,
+			extCommitCodec,
+			app.Logger(),
+		),
+		oracleMetrics,
+	)
+	app.SetExtendVoteHandler(voteExtensionsHandler.ExtendVoteHandler())
+	app.SetVerifyVoteExtensionHandler(voteExtensionsHandler.VerifyVoteExtensionHandler())
+}
+
+type AppUpgrade struct {
+	Name         string
+	Handler      upgradetypes.UpgradeHandler
+	StoreUpgrade storetypes.StoreUpgrades
+}
+
+// createSlinkyUpgrader returns the upgrade name and an upgrade handler that:
+// - runs migrations
+// - updates the consensus keeper params with a vote extension enable height. (height of upgrade + 10).
+// - adds the core markets to x/marketmap keeper.
+// additionally, it returns the required StoreUpgrades needed for the new slinky modules added to this chain.
+func createSlinkyUpgrader(app *App) AppUpgrade {
+	return AppUpgrade{
+		Name: "TODO: NAME THIS UPGRADE",
+		Handler: func(ctx context.Context, _ upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+			migrations, err := app.ModuleManager.RunMigrations(ctx, app.Configurator(), fromVM)
+			if err != nil {
+				return nil, err
+			}
+			// upgrade consensus params to enable vote extensions
+			consensusParams, err := app.ConsensusParamsKeeper.Params(ctx, nil)
+			if err != nil {
+				return nil, err
+			}
+			sdkCtx := sdk.UnwrapSDKContext(ctx)
+			consensusParams.Params.Abci = &tmtypes.ABCIParams{
+				VoteExtensionsEnableHeight: sdkCtx.BlockHeight() + int64(10),
+			}
+			_, err = app.ConsensusParamsKeeper.UpdateParams(ctx, &consensustypes.MsgUpdateParams{
+				Authority: app.ConsensusParamsKeeper.GetAuthority(),
+				Block:     consensusParams.Params.Block,
+				Evidence:  consensusParams.Params.Evidence,
+				Validator: consensusParams.Params.Validator,
+				Abci:      consensusParams.Params.Abci,
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			// add core markets
+			coreMarkets := constants.CoreMarketMap
+			markets := coreMarkets.Markets
+			keys := make([]string, 0, len(markets))
+			for name, _ := range markets {
+				keys = append(keys, name)
+			}
+			slices.Sort(keys)
+
+			// iterates over slice and not map
+			for _, marketName := range keys {
+				// create market
+				market := markets[marketName]
+				err = app.MarketMapKeeper.CreateMarket(sdkCtx, market)
+				if err != nil {
+					return nil, err
+				}
+
+				// invoke hooks
+				err = app.MarketMapKeeper.Hooks().AfterMarketCreated(sdkCtx, market)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			err = app.MarketMapKeeper.SetParams(
+				sdkCtx,
+				marketmaptypes.Params{
+					Admin:             "TODO: YOUR ADMIN ADDRESS", // this can be set to governance, or an address of your choosing.
+					MarketAuthorities: []string{"warden1ua63s43u2p4v38pxhcxmps0tj2gudyw2rctzk3"}},
+			)
```

Comment:
i can set `Admin` to the gov address, or some other address. just let me know the address if you want me to include that in this PR.

Reply:
I think we could set this up in a governance to test how it works as well.

Reply:
yes, let's go with gov

Reply:
[make governance address the marketmap admin, as well as an authority.…](https://github.com/warden-protocol/wardenprotocol/pull/425/commits/e8886741d5332c77f9175cfdfd6e67263bc10b45)

added gov as admin and authority here.